zerver/lib/message.py

```diff
@@ -242,6 +246,29 @@ def visible_edit_history_for_message(
     return visible_edit_history
 
 
+def topic_resolve_toggled(topic: str, prev_topic: str) -> bool:
```

Comment:
Because we have the non-truncated topic when we build the message edit request, it seemed better to leave that logic as is and make a separate helper function here for figuring out if the resolved/unresolved logic when we only have the truncated resolved topic in the edit history.

Reply:
Probably good to have this in a docstring/comment here, so one isn't tempted to deduplicate without knowing the reasoning.

Reply:
Added that comment above.